cmd/kubeadm/app/cmd/join.go

```diff
@@ -169,6 +195,12 @@ func AddJoinConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiext.NodeConfigurat
 		featureGatesString, "feature-gates", *featureGatesString,
 		"A set of key=value pairs that describe feature gates for various features. "+
 			"Options are:\n"+strings.Join(features.KnownFeatures(&features.InitFeatureGates), "\n"))
+	flagSet.BoolVar(
+		&cfg.Master, "master", cfg.Master,
+		"Join as a master node")
+	flagSet.StringVar(
+		&cfg.MasterConfigurationFile, "masterConfiguration", cfg.MasterConfigurationFile,
```

Comment:
I'm still confused about the MasterConfigurationFile vs. grabbing the configmap.

Reply:
see notes above